### DISCUSSION TITLE
It seems there are some differences between shadowsocks aead 2022 and the implementation from shadowsocks-rust

### DISCUSSION THREAD

**Discussion by passingwang** [shadowsocks/shadowsocks-rust#1615] [Q&A] [answered]
I didn't find any code about the 2022-blake3-chacha*-poly1305 cipher in `crates/shadowsocks/src/relay/tcprelay/aead_2022.rs`. In `crates/shadowsocks/src/relay/udprelay/aead_2022.rs`, Shadowsocks-Rust seems using ChaCha-Poly1305 instead of XChaCha-Poly1305 as described in shadowsocks.org. The third-party implementation sing-shadowsocks also using ChaCha instead of XChaCha. Btw, the value of `AEAD2022_EIH_SUBKEY_CONTEXT`, which used for Blake3 key derivation (from the name of this variable, it seems related to EIN) is `shadowsocks 2022 identity subkey`, not the `shadowsocks 2022 session subkey` as described in the documentation.

**Comment by database64128** [comment, marked as answer]
The implementation of Shadowsocks 2022 in shadowsocks-rust is correct. I wrote the spec and reviewed the code.

**Reply by passingwang** [reply]
> The implementation of Shadowsocks 2022 in shadowsocks-rust is correct. I wrote the spec and reviewed the code.

Thanks for your anwser.
